### PR TITLE
Rename CPUProfiler to ProfilerBase in Metal runtime to fix the build

### PR DESCRIPTION
**Pull request by k-ye**

### CODE REVIEW

File: taichi/profiler.cpp [hunk 1]
```diff
@@ -139,6 +139,7 @@ class CPUProfiler : public ProfilerBase {
 
 std::unique_ptr<ProfilerBase> make_profiler(Arch arch) {
   if (arch == Arch::x86_64 || arch == Arch::arm) {
+    // TODO: Arch::metal also uses CPUProfiler, rename it to DefaultProfiler?
```
Comment (yuanming-hu) [inline taichi/profiler.cpp]: Good idea!

**Comment by yuanming-hu** [comment]
Sorry for breaking your code! :~)

File: taichi/profiler.cpp (hunk 1 above)
Comment (yuanming-hu) [reply taichi/profiler.cpp]: [master c5e0da63] CPUProfiler->DefaultProfiler; make_profiler now works for Arch::metal

Comment (k-ye) [reply taichi/profiler.cpp]: Nice!

**Pull request merged**